### ISSUE TITLE
Display localized card assets in Twitch HDT extension

### ISSUE THREAD

**Issue by beheh** [HearthSim/twitch-hdt-frontend#8]
As a user viewing a Twitch Hearthstone stream in which the streamer has configured the Twitch HDT extension, I would like to see card metadata / assets in my native language, so that I can better enjoy the game that I am watching.

This story is an attempt to make some small localization enhancements to the user experience without making major investment in localizing the Twitch extension. It seems like this is possible by making `react-hs-components` locale-aware and pull the appropriate assets from hearthstonejson.com.

This story is complete when:
- [x] The Card component in `react-hs-components` can be configured (via a property) with a "Hearthstone locale," e.g., "enUS" or "frFR"
- [x] The Twitch HDT extension accepts the viewer's preferred language (supplied via Twitch) and maps it to a Hearthstone locale, falling back to "enUS" when the language cannot be detected or mapped.
- [x] The extension initializes `hearthstonejson-client` with the configured locale
- [x] The extension renders the `Card` and `CardTile` components by passing the respective language or data in the language to the components
- [x] These enhancements are covered by unit tests

**Comment by joolean** [comment]
Closing: Low urgency.

**Comment by beheh** [comment]
The Twitch extension has no access to HTTP headers, as it's a client-side-only application. Twitch supplies the broadcaster's and the viewer's language though.

This ticket is incomplete. It also needs to describe how we configure `hearthstonejson-client` with the correct language when called from the extension, so that `CardTile`s are localized as well.

**Comment by jleclanche** [comment]
Yes we'd be using the viewer's language for this. (And the broadcaster's if we ever localize the settings)

**Comment by joolean** [comment]
I've made a few changes to the acceptance criteria but I'm not sure I understand your concern about CardTiles, @beheh. Feel free to make any tweaks you want to the description.

**Comment by beheh** [comment]
A Card is the actual card image as it appears ingame, whereas a card tile is the square box we use on HSReplay.net, the Deck Trackers and the Twitch Extension for cards in a deck list.
The extension uses both: CardTiles overlayed on the stream, and Cards when you hover over one of them (just like on HSReplay.net).

I'll add them to this story.

**Comment by jleclanche** [comment]
Twitch languages:

```py
('ar', 'العربية')
('bg', 'Български')
('cs', 'Čeština')
('da', 'Dansk')
('de', 'Deutsch')
('el', 'Ελληνικά')
('en', 'English')
('en-gb', 'English - UK')
('es', 'Español - España')
('es-mx', 'Español - Latinoamérica')
('fi', 'Suomi')
('fr', 'Français')
('hi', 'हिदी')
('hu', 'Magyar')
('it', 'Italiano')
('ja', '日本語')
('ko', '한국어')
('nl', 'Nederlands')
('no', 'Norsk')
('pl', 'Polski')
('pt', 'Português')
('pt-br', 'Português - Brasil')
('ro', 'Română')
('ru', 'Русский')
('sk', 'Slovenčina')
('sv', 'Svenska')
('th', 'ภาษาไทย')
('tr', 'Türkçe')
('vi', 'Tiếng Việt')
('zh-cn', '中文 简体')
('zh-tw', '中文 繁體')
```

The ones we care about:

```py
('de', 'Deutsch')
('en', 'English')
('es', 'Español - España')
('es-mx', 'Español - Latinoamérica')
('fr', 'Français')
('it', 'Italiano')
('ja', '日本語')
('ko', '한국어')
('pl', 'Polski')
('pt-br', 'Português - Brasil')
('ru', 'Русский')
('th', 'ภาษาไทย')
('zh-cn', '中文 简体')
('zh-tw', '中文 繁體')

# Additional for the mapping:
('en-gb', 'English - UK')
('pt', 'Português')
```

Map:
```json
{
"de": "deDE",
"en": "enUS",
"en-gb": "enUS",
"es": "esES",
"es-mx": "esMX",
"fr": "frFR",
"it": "itIT",
"ja": "jaJP",
"ko": "koKR",
"pl": "plPL",
"pt": "ptBR",
"pt-br": "ptBR",
"ru": "ruRU",
"th": "thTH",
"zh-cn": "zhCN",
"zh-tw": "zhTW",
}
```

We need to decide whether we're getting the language out of the stream's language or the viewer's language. I have a `en` UI and I'm watching an `fr` stream, should the cards show up in french or english? Stream UI makes more sense for consistency with (potentially) the game, viewer's UI is more reliable and makes more sense for consistency with the Twitch UI.

We should consider having a language picker in the option menu either way.

**Comment by jleclanche** [comment]
CardTile will not have a locale parameter because it doesn't handle json loading by itself, it gets a name from downstream.

**Comment by beheh** [comment]
Released in 0.4.0.
